### ISSUE TITLE
Integer value parsed incorrectly in some cases

### ISSUE THREAD

**Issue by burmisov** [stephenwvickers/node-net-snmp#5]
Hi! Thanks for an awesome module!

I have a problem with a rather exotic device trying to read a temperature sensor value.
The code is simple:
```js
var snmp = require('net-snmp');
var session = snmp.createSession('10.157.207.28', 'public', { version: snmp.Version1 });
session.get(['1.3.6.1.4.1.40418.2.4.4.1.0'], function (error, varbinds) {
	if (error) {
		throw error;
	}
	console.log('result>', varbinds);
	session.close();
});
```

What I get is an `Integer too long (129)` error. I drilled it down to the source and it looks like at [this point](https://github.com/stephenwvickers/node-net-snmp/blob/master/index.js#L216-L220) an extra byte with value 129 is getting in the way for some reason. If I discard it prior to processing (i.e., add another `buffer.readByte ();` after [this line](https://github.com/stephenwvickers/node-net-snmp/blob/master/index.js#L217), the value is parsed correctly. But I guess that'a a pretty bad hack.

I thought maybe the device is causing problems (I can read multiple other devices without issues), but both `snmpget` cli utility and the `Free SNMP MIB Browser` of ManageEngine seem to handle this OID of my device without any issues.

Maybe I will get down to the problem myself eventually, but some couple hours spent to this moment didn't bring me a solution.

**Comment by stephenwvickers** [comment]
Hi Sergey,

Sorry for the delay in my response.

This would suggest an issue with the structure of the response packet.

Are you able to send me a Wireshark capture which includes the SNMP response packet (or similar)?

Thanks

Steve

**Comment by burmisov** [comment]
Hi Stephen!
No worries, my fork with a hack that checks for the `129` value surprisingly works for all of the devices I have at hand :)
I guess there may be wrong structure in the response,  I never would create and issue here but for snpmget getting a good result.
I'll create a capture ASAP, wen I get my hands on the device again.

**Comment by stephenwvickers** [comment]
That's ok, I understand.

If there is an issue there I would like to address it for you.

Thanks

Steve

**Comment by stephenwvickers** [comment]
If you would like to look into this further just reopen the issue and I will continue to investigate.

Thanks

Steve

**Comment by markabrahams** [comment]
This is long closed, but @burmisov if you're still out there, and you still care, this issue is fixed in version 2.9.8 of the npm.  See #129 for further details of the problem and fix.
